site/app/views/submission/HomeworkView.php

```diff
@@ -315,13 +315,29 @@ private function renderSubmitBox(Gradeable $gradeable, $graded_gradeable, $versi
                     ]
                 );
             }
-            if ($graded_gradeable !== null) {
-                $notebook_data = $notebook_model->getMostRecentNotebookSubmissions(
-                    $graded_gradeable->getAutoGradedGradeable()->getHighestVersion(),
-                    $notebook,
-                    $this->core->getUser()->getId()
-                );
+            if ($version_instance) {
+                if ($graded_gradeable !== null) {
+                    $notebook_data = $notebook_model->getMostRecentNotebookSubmissions(
+                        $graded_gradeable->getAutoGradedGradeable()->getHighestVersion(),
+                        $notebook,
+                        $this->core->getUser()->getId(),
+                        $version_instance->getVersion() ?? null,
```

Comment:
Can `getVersion()` return something other than `?int` that we need the null coaelescing operator here?

Reply:
Ignoring the comment above, why have the null coalescing operator here? What is it doing?

Reply:
looks like it's doing nothing, I didn't write that line so I didn't look into it before.